src/app/gateways/views/BuiltinGatewayDetailView.vue

```diff
@@ -18,29 +18,19 @@
       }"
     >
       <AppView>
-        <DataSource
+        <DataLoader
           v-slot="{ data, error }: MeshGatewaySource"
           :src="`/meshes/${route.params.mesh}/mesh-gateways/${route.params.gateway}`"
         >
-          <ErrorBlock
-            v-if="error"
-            :error="error"
-          />
-
-          <LoadingBlock v-else-if="data === undefined" />
-
-          <KCard v-else>
-            <DataSource
-              v-slot="{ data: dataplanesData, error: dataplanesError }: DataplaneOverviewCollectionSource"
-              :src="`/meshes/${route.params.mesh}/dataplanes/for/${data.selectors[0].match['kuma.io/service']}?page=${route.params.page}&size=${route.params.size}&search=${route.params.s}`"
-            >
-              <ErrorBlock
-                v-if="dataplanesError !== undefined"
-                :error="dataplanesError"
-              />
-
+          <DataLoader
+            v-if="data"
+            v-slot="{ data: dataplanesData, error: dataplanesError }: DataplaneOverviewCollectionSource"
+            :src="`/meshes/${route.params.mesh}/dataplanes/for/${data.selectors[0].match['kuma.io/service']}?page=${route.params.page}&size=${route.params.size}&search=${route.params.s}`"
```

Comment:
I just noticed this, but is it ok to just use `selectors[0]`, can we have selected dataplanes at `selectors[1]` etc do you know?

Reply:
As far as I know, this should be safe, but I’m also not certain (cc @lahabana).

If we need to consider multiple selectors, that eithers means we’d expect multiple Dataplane lists or alternatively we would need to query the Dataplane API with multiple `kuma.io/service` tag filter parameters (does the API handle that? From a quick test this doesn’t seem to be the case: `curl http://localhost:5681/meshes/default/dataplanes\?tag\=kuma.io/service:redis_kuma-demo_svc_6379\&tag\=kuma.io/service:demo-app_kuma-demo_svc_5000` only takes the last tag filter into account).

Reply:
FYI I opened https://github.com/kumahq/kuma/issues/9481 and we’ll settle this there.